plebnet/agent/core.py

```diff
@@ -108,9 +115,11 @@ def check():
 
     # These need a matchmaker, otherwise agent will be stuck waiting.
     if market_controller.has_matchmakers():
-        if config.time_to_expiration() <= plebnet_settings.TIME_IN_DAY:
-            update_offer()
-            attempt_purchase()
+        # pass
+        strategies[plebnet_settings.get_instance().strategy_name()]().apply()
+        # if config.time_to_expiration() <= plebnet_settings.TIME_IN_DAY:
+            # update_offer()
+            # attempt_purchase()
```

Comment:
Please remove commented code. Also, do you understand why do we need to check for matchmakers here? Wouldn't it be better to check that only when placing offer? When we already have enough BTC and just want to `attempt_purchase`, we do not need any matchmakers.

Reply:
Oops.

From what I understood, you can only create offers if there are matchmakers, so there is no point in trying to apply a strategy if you can't place offers on the market